packages/mui-base/src/BadgeUnstyled/BadgeUnstyled.js

```diff
@@ -42,36 +43,17 @@ const BadgeUnstyled = React.forwardRef(function BadgeUnstyled(props, ref) {
     overlap: overlapProp = 'rectangular',
```

Comment:
I'm wondering if it makes sense to include this prop as it's not used in the unstyled component.

Reply:
Good catch, looks like we can move it to the styled component’s props.

Reply:
Done with https://github.com/mui-org/material-ui/pull/30246/commits/7674166ce6ceb7bf5e4563b621cb788ebcf416d5